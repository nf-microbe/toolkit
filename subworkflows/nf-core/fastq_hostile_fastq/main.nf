// Import modules
include { HOSTILE_FETCH } from '../../../modules/nf-core/hostile/fetch/main'
include { HOSTILE_CLEAN } from '../../../modules/nf-core/hostile/clean/main'

// Run subworkflow
workflow FASTQ_HOSTILE_FASTQ {

    take:
    fastq_gz        // channel: [ [ meta ], [ reads_1.fastq.gz, reads_2.fastq.gz ] ] (MANDATORY)
    hostile_index   // val: /path/to/host_index.bt2 (MANDATORY)

    main:
    def ch_versions = Channel.empty()


    //
    // MODULE: Download hostile reference
    //
    HOSTILE_FETCH(
        Channel.of([ [ id:'hostile_index' ], hostile_index ])
    )
    def ch_hostile_index    = HOSTILE_FETCH.out.index.first()
    ch_versions             = ch_versions.mix(HOSTILE_FETCH.out.versions)

    //
    // MODULE: Remove human reads
    //
    HOSTILE_CLEAN(
        fastq_gz,
        ch_hostile_index
    )
    ch_versions = ch_versions.mix(HOSTILE_CLEAN.out.versions)

    emit:
    fastq_gz    = HOSTILE_CLEAN.out.fastq   // channel: [ val(meta), [ reads_1.fastq.gz, reads_2.fastq.gz ] ]
    versions    = ch_versions               // channel: [ software_versions.yml ]
}

