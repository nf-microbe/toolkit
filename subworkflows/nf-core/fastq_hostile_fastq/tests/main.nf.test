nextflow_workflow {

    name "Test Subworkflow FASTQ_HOSTILE_FASTQ"
    script "../main.nf"
    workflow "FASTQ_HOSTILE_FASTQ"

    tag "subworkflows"
    tag "subworkflows_nfcore"
    tag "subworkflows/fastq_hostile_fastq"
    tag "hostile/fetch"
    tag "hostile/clean"

    test("human-t2t-hla.argos-bacteria-985_rs-viral-202401_ml-phage-202401") {
        when {
            workflow {
                """
                input[0] = Channel.of(
                    [
                        [ id:'test1', single_end:false ], // meta map
                        [
                            file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz',checkIfExists: true),
                            file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_2.fastq.gz',checkIfExists: true)
                        ]
                    ],
                    [
                        [ id:'test2', single_end:false ], // meta map
                        [
                            file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test2_1.fastq.gz',checkIfExists: true),
                            file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test2_2.fastq.gz',checkIfExists: true)
                        ]
                    ]
                )
                input[1] = [
                    [ id:'reference' ],
                    "human-t2t-hla.argos-bacteria-985_rs-viral-202401_ml-phage-202401"
                ]
                """
            }
        }
        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
            )
        }
    }
}
