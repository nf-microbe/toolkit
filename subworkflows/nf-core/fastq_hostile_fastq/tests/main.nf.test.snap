{
    "human-t2t-hla.argos-bacteria-985_rs-viral-202401_ml-phage-202401": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test1",
                            "single_end": false
                        },
                        [
                            "test_1.clean_1.fastq.gz:md5,00682af915b1267f83d781d0d1b59cd7",
                            "test_2.clean_2.fastq.gz:md5,44013d83b4925719051e0f4f79987c5a"
                        ]
                    ],
                    [
                        {
                            "id": "test2",
                            "single_end": false
                        },
                        [
                            "test2_1.clean_1.fastq.gz:md5,00682af915b1267f83d781d0d1b59cd7",
                            "test2_2.clean_2.fastq.gz:md5,44013d83b4925719051e0f4f79987c5a"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,1ed458e555348606a9d2d39d2a1881ab",
                    "versions.yml:md5,978fe2c891e3081efc0966510abbf194",
                    "versions.yml:md5,978fe2c891e3081efc0966510abbf194"
                ],
                "fastq_gz": [
                    [
                        {
                            "id": "test1",
                            "single_end": false
                        },
                        [
                            "test_1.clean_1.fastq.gz:md5,00682af915b1267f83d781d0d1b59cd7",
                            "test_2.clean_2.fastq.gz:md5,44013d83b4925719051e0f4f79987c5a"
                        ]
                    ],
                    [
                        {
                            "id": "test2",
                            "single_end": false
                        },
                        [
                            "test2_1.clean_1.fastq.gz:md5,00682af915b1267f83d781d0d1b59cd7",
                            "test2_2.clean_2.fastq.gz:md5,44013d83b4925719051e0f4f79987c5a"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,1ed458e555348606a9d2d39d2a1881ab",
                    "versions.yml:md5,978fe2c891e3081efc0966510abbf194",
                    "versions.yml:md5,978fe2c891e3081efc0966510abbf194"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-11-18T08:46:14.876439637"
    }
}